### PR TITLE
STAF-33: Make a mobile page layout for protected pages 

### PR DESCRIPTION
**Pull request by RomanLavrunov**
navbar used for mockup purposes only and should be rebuilt with the next stages of development.

Please check if the following use of decimals in vh is applicable
root-layout-container => grid-template-rows: 7.5vh 84vh 8.5vh;

### CODE REVIEW

File: src/core/components/RootNavLayout.css [hunk 1]
```diff
@@ -0,0 +1,39 @@
+.root-layout-container {
+  position: fixed;
+  overflow: auto;
+  width: 100%;
+  display: grid;
+  grid-template-rows: 7.5vh 84vh 8.5vh;
```
Comment (devmc-ee) [inline src/core/components/RootNavLayout.css:6]: Let's set the height of header and nav-bar areas with px or rem units, as it should not change on different screens sizes, unless we adjust it with  @media
The middle area could be set to 1fr as it should fill all the available for it area

File: src/core/components/RootNavLayout.css [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  height: 100%;
```
Comment (devmc-ee) [inline src/core/components/RootNavLayout.css:7]: We dont need to use height if the position is fixed and it is set with top and bottom values. Top 0 and bottom 0  are equal to 100vh

File: src/core/components/RootNavLayout.tsx [hunk 3]
```diff
@@ -1,11 +1,15 @@
 import { Outlet } from 'react-router-dom';
 import { NavMenu } from './NavMenu';
+import './RootNavLayout.css';
```
Comment (devmc-ee) [inline src/core/components/RootNavLayout.tsx:3]: Please make a new folder for the RootNavLayout component and move there the both files (.tsx and css). Please also add the index.ts and make a named re-export for the component, that should simplify the import

File: src/core/components/NavMenu.tsx [hunk 4]
```diff
@@ -2,6 +2,13 @@ import { Link, useLocation } from 'react-router-dom';
 import { useAppDispatch } from '../../config/redux/hooks';
 import { clearUserData } from '../../user/redux/userSlice';
 import { routes } from '../routing/routes';
+import { CustomButton } from '../components/buttons/Button/Button';
+import HomeIcon from '@mui/icons-material/Home';
+import CalendarMonthIcon from '@mui/icons-material/CalendarMonth';
+import LogoutIcon from '@mui/icons-material/Logout';
+import { useState, MouseEvent } from 'react';
+import { PendingActions } from '@mui/icons-material';
+import './NavMenu.css';
```
Comment (devmc-ee) [inline src/core/components/NavMenu.tsx:11]: Please make a new folder for the NavMenu component and move there the both files (.tsx and css). Please also add the index.ts and make a named re-export for the component, that should simplify the import

**Review by devmc-ee — CHANGES_REQUESTED** [review]

File: src/core/components/RootNavLayout.tsx (hunk 3 above)
Comment (RomanLavrunov) [reply src/core/components/RootNavLayout.tsx]: Done!

File: src/core/components/NavMenu.tsx (hunk 4 above)
Comment (RomanLavrunov) [reply src/core/components/NavMenu.tsx]: Done!

File: src/core/components/RootNavLayout.css (hunk 2 above)
Comment (RomanLavrunov) [reply src/core/components/RootNavLayout.css]: Got it! Fixed!

File: src/core/components/RootNavLayout.css (hunk 1 above)
Comment (RomanLavrunov) [reply src/core/components/RootNavLayout.css]: Ok! I Canged header and footer to 3.75 and 4.19 rem respectively.

File: src/App.tsx [hunk 5]
```diff
@@ -4,7 +4,7 @@ import { Dashboard, Login } from './pages';
 import { ProtectedRoute } from './core/components/ProtectedRoute';
 import { routes } from './core/routing/routes';
 import { NewStaffPage, StaffPage } from './staff/components';
-import { RootNavLayout } from './core/components/RootNavLayout';
+import { RootNavLayout } from './core/components/RootNavLayout/RootNavLayout';
```
Comment (devmc-ee) [inline src/App.tsx:7]: This import can be shortened as there is an index file in the RootNavLayout folder with the named re-exoport

File: src/core/components/RootNavLayout/RootNavLayout.tsx [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+import './RootNavLayout.css';
+
+export const RootNavLayout = (): JSX.Element => {
+  return (
+    <div className="root-layout-container">
+      <div className="root-layout-header"></div>
+      <div className="root-content-container">
+        <Outlet />
+      </div>
+      <NavMenu />
```
Comment (devmc-ee) [inline src/core/components/RootNavLayout/RootNavLayout.tsx:12]: Please add a container for the footer area. This container should define area dimensions and location. The footer area might contain more than 1 component

**Review by devmc-ee — CHANGES_REQUESTED** [review]

File: src/App.tsx (hunk 5 above)
Comment (RomanLavrunov) [reply src/App.tsx]: Import fixed.

File: src/core/components/RootNavLayout/RootNavLayout.tsx (hunk 6 above)
Comment (RomanLavrunov) [reply src/core/components/RootNavLayout/RootNavLayout.tsx]: Footer container added.

**Review by devmc-ee — APPROVED** [review]

**Pull request merged**